src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +10,30 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User can't be null!");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null!");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can't be null!");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age can't be null!");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must be at least 6 characters long!");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new RegistrationException("Password must be at least 6 characters long!");
+        }
+        if (user.getAge() < 18) {
+            throw new RegistrationException("User must be at least 18 years old!");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User already exists!");
+        }
+        return storageDao.add(user);
```

Comment:
Maybe it's will be better to separate checks by field in different methods, and don't use every checks in one method